sdk/Metrics/ValueRecorder.php

```diff
@@ -0,0 +1,161 @@
+ *
+ * Example: One of the most common uses for ValueRecorder is to capture latency
+ * measurements. Latency measurements are not additive in the sense that
+ * there is little need to know the latency-sum of all processed requests.
+ * We use a ValueRecorder instrument to capture latency measurements
+ * typically because we are interested in knowing mean, median, and other
+ * summary statistics about individual events.
+ *
+ * The default aggregation for ValueRecorder computes the minimum and maximum
+ * values, the sum of event values, and the count of events, allowing the rate,
+ * the mean, and range of input values to be monitored.
+ */
+class ValueRecorder extends AbstractMetric implements API\ValueRecorder, API\LabelableMetric
+{
+    use HasLabels;
+
+    /**
+     * @var float $valueSum
+     * @var float $valueMin
+     * @var float $valueMax
+     * @var int $valueCount
+     */
+    protected $valueSum = 0;
+    protected $valueMin = INF;
+    protected $valueMax = -INF;
+    protected $valueCount = 0;
+
+    /*
+     * Testing floating point values for equality is problematic, due to
+     * the way that they are represented internally.  Therefore, force all
+     * precision to a predetermined number, and equality can be determined.
+     */
+    private $decimalPointPrecision = 10;
+
+    /**
+     * getType: get the type of metric instrument
+     *
+     * @return  int
+     */
+    public function getType(): int
+    {
+        return API\MetricKind::VALUE_RECORDER;
+    }
+
+    /**
+     * Returns the sum of the values
+     *
+     * @access	public
+     * @return	float
+     */
+    public function getValueSum(): float
+    {
+        return $this->valueSum;
+    }
+
+    /**
+     * Returns the min of the values
+     *
+     * @access	public
+     * @return	float
+     */
+    public function getValueMin(): float
+    {
+        return $this->valueMin;
+    }
+
+    /**
+     * Returns the max of the values
+     *
+     * @access	public
+     * @return	float
+     */
+    public function getValueMax(): float
+    {
+        return $this->valueMax;
+    }
+
+    /**
+     * Returns the mean of the values
+     *
+     * @access	public
+     * @return	float
+     */
+    public function getValueMean(): float
+    {
+        if (0 == $this->valueCount) {
+            return 0;
+        }
+
+        return ($this->valueSum/$this->valueCount);
+    }
+
+    /**
+     * Returns the count of the values
+     *
+     * @access	public
+     * @return	int
+     */
+    public function getValueCount(): int
+    {
+        return $this->valueCount;
+    }
+
+    /**
+     * Updates the ValueRecorder's value with the specified value then returns
+     * the current value count.
+     *
+     * @access	public
+     *
+     * @param float $value, accepts INTs or FLOATs. If value is an int, it it cast as a float.
+     *
+     * @return int $valueCount, returns the current count of values recorded.
```

Comment:
Looks like there is an inconsistency here because the method `record` doesn't actually return anything